dash-renderer/src/actions/setAppReadyState.js

```diff
@@ -1,87 +1,43 @@
-import {filter} from 'ramda';
-import {createAction} from 'redux-actions';
+import { path } from 'ramda';
+import { createAction } from 'redux-actions';
 
-import isSimpleComponent from '../isSimpleComponent';
 import Registry from './../registry';
-import {getAction} from './constants';
-import {isReady} from '@plotly/dash-component-plugins';
+import { getAction } from './constants';
+import { isReady } from '@plotly/dash-component-plugins';
 
-const isAppReady = layout => {
-    const queue = [layout];
-
-    const components = {};
-    const ids = {};
-
-    /* Would be much simpler if the Registry was aware of what it contained... */
-    while (queue.length) {
-        const elementLayout = queue.shift();
-        if (!elementLayout) {
-            continue;
-        }
-
-        const id = elementLayout.props && elementLayout.props.id;
-        const children = elementLayout.props && elementLayout.props.children;
-        const namespace = elementLayout.namespace;
-        const type = elementLayout.type;
-
-        components[namespace] = components[namespace] || {};
-        components[namespace][type] = type;
+const isAppReady = (layout, paths) => targets => {
+    if (!layout || !paths || !Array.isArray(targets)) {
+        return true;
```

Comment:
Is there a use case for this? Feels like an error rather than a silent success.

Reply:
Relates to the comment below. Refactoring to take `isAppReady` outside of the state, timing issues will be moot.